compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -35,30 +113,39 @@ fn pre_link_args(os: &'static str, arch: &'static str, abi: &'static str) -> Lin
     args
 }
 
-pub fn opts(os: &'static str, arch: &'static str, abi: &'static str) -> TargetOptions {
-    // ELF TLS is only available in macOS 10.7+. If you try to compile for 10.6
-    // either the linker will complain if it is used or the binary will end up
-    // segfaulting at runtime when run on 10.6. Rust by default supports macOS
-    // 10.7+, but there is a standard environment variable,
-    // MACOSX_DEPLOYMENT_TARGET, which is used to signal targeting older
-    // versions of macOS. For example compiling on 10.10 with
-    // MACOSX_DEPLOYMENT_TARGET set to 10.6 will cause the linker to generate
-    // warnings about the usage of ELF TLS.
-    //
-    // Here we detect what version is being requested, defaulting to 10.7. ELF
-    // TLS is flagged as enabled if it looks to be supported. The architecture
-    // only matters for default deployment target which is 11.0 for ARM64 and
-    // 10.7 for everything else.
-    let has_thread_local = macos_deployment_target("x86_64") >= (10, 7);
+pub fn opts(os: &'static str, arch: Arch) -> TargetOptions {
+    let has_thread_local = if os == "macos" {
+        // ELF TLS is only available in macOS 10.7+. If you try to compile for 10.6
```

Comment:
You didn't write this but it might be worth changing this to say "static TLS", since obviously ELF is not used by these targets (which use Mach-O not ELF).

Up to the actual reviewer.

Reply:
Was easy enough to change around so "ELF TLS" has gone away